### DISCUSSION TITLE
Using a media picker in a repeater

### DISCUSSION THREAD

**Discussion by freshleafmedia** [area17/twill#1546] [Q&A]
Hi, Is it possible to have a media picker in a repeater, outside of a block? For example:

`form.blade.php`
```
@extends('twill::layouts.form')

@formField('repeater', [
    'type' => 'imageRepeater',
])
```

---

`imageRepeater.blade.php`
```
@twillRepeaterTitle('Image')

@formField('medias', [
    'label' => 'Image',
    'name' => 'image',
])
```

**Comment by pboivin** [comment]
Hi @freshleafmedia, yes this is possible.

Expanding on the basic example in the docs:
https://twill.io/docs/form-fields/repeater.html#using-repeater-fields

You could define your media role on the `TeamMember` model:
```
    public $mediasParams = [
        'profile' => [
            'default' => [
                [
                    'name' => 'default',
                    'ratio' => 1,
                ],
            ],
        ],
    ];
```

And then add the `medias` field to the `team-member.blade.php` repeater template:
```
@formField('medias', [
    'name' => 'profile',
    'label' => 'Profile',
])
```

I don't think any other config is needed to make this work. Are you getting any errors trying to implement this in your app?

**Reply by freshleafmedia** [reply]
Hi @pboivin,

I don't get any errors, the media just gets silently ignored on save.

My use case is slightly different from the example because I don't have a requirement for an equivalent to the `TeamMember` model. I want to attach a variable number of media directly to the `Team` model.

It appears that another model is required however, is that correct?

Thanks!

**Reply by freshleafmedia** [reply]
Just re-read the docs and all I needed was to add the `max` attribute to the media `formField`:

```
@formField('medias', [
    'name' => 'profile',
    'label' => 'Profile',
    'max' => 3,
])
```
